core/commands/object/object.go

```diff
@@ -234,7 +237,19 @@ This command outputs data in the following encodings:
 	Type: Node{},
 	Marshalers: cmds.MarshalerMap{
 		cmds.Protobuf: func(res cmds.Response) (io.Reader, error) {
-			node := res.Output().(*Node)
+			var (
+				ch <-chan interface{}
+				ok bool
+			)
+			ch, ok = res.Output().(chan interface{})
+			if !ok {
+				ch, ok = res.Output().(<-chan interface{})
+				if !ok {
+					panic("object.get marshaler: res.Output() is not chan interface{}")
+				}
+			}
+
+			node := (<-ch).(*Node)
```

Comment:
Because I don't check the type? I guess I need to find a general answer for this...ErrCast is also stupid